### DISCUSSION TITLE
[Support] Cannot read properties of undefined (reading 'debug')

### DISCUSSION THREAD

**Discussion by hooliy** [midwayjs/midway#3538] [Q&A]
### Describe the problem（描述问题）

"bundle_start": "cross-env NODE_ENV=production node ./build/index.js"

单文件打包，启动提示 Cannot read properties of undefined (reading 'debug')

### Midway Versions（Midway 版本）

 @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.13.7)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.13.7)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.13.7)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.13.7)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.13.9)
✓ @midwayjs/info(3.13.9)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.13.9)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.13.9)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.13.9)
✓ @midwayjs/version(3.13.9)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.13.9)
✓ @midwayjs/ws(not installed)

**Comment by czy88840616** [comment]
堆栈呢？

**Reply by hooliy** [reply]
mac下没发现此问题，windows 下重现步骤：
1、脚手架生成新项目
2、按 [单文件部署文档](https://www.midwayjs.org/docs/deployment#%E5%8D%95%E6%96%87%E4%BB%B6%E6%9E%84%E5%BB%BA%E9%83%A8%E7%BD%B2) 修改。
3、运行报错。

$ cross-env NODE_ENV=production node ./build/index.js
2024-01-08 12:26:15.426 ERROR 16576 TypeError: Cannot read properties of undefined (reading 'debug')
    at KoaControllerGenerator.loadMidwayController (D:\works\etu\build\index.js:6446:20)
    at async MidwayKoaFramework.loadMidwayController (D:\works\etu\build\index.js:18154:9)
    at async MidwayKoaFramework.run (D:\works\etu\build\index.js:18178:9)
    at async MidwayFrameworkService.runFramework (D:\works\etu\build\index.js:11670:17)
    at async MidwayLifeCycleService.init (D:\works\etu\build\index.js:11996:9)
    at async ObjectCreator.doInitAsync (D:\works\etu\build\index.js:9907:17)
    at async ManagedResolverFactory.createAsync (D:\works\etu\build\index.js:7515:9)
    at async initializeGlobalApplicationContext (D:\works\etu\build\index.js:13471:5)
    at async BootstrapStarter.init (D:\works\etu\build\index.js:3596:35)
    at async BootstrapStarter.run (D:\works\etu\build\index.js:3604:35)
2024-01-08 12:26:15.429 INFO 16576 [midway:bootstrap] exit with code:1

**Reply by czy88840616** [reply]
(D:\works\etu\build\index.js:6446:20) 这行是什么？

**Reply by hooliy** [reply]
<img width="720" alt="image" src="https://github.com/midwayjs/midway/assets/3368691/f92bc178-5c11-4cf5-9d99-5873fa5c493b">

**Reply by czy88840616** [reply]
package.json 中没有引用 @midwaysj/logger  ？

**Reply by hooliy** [reply]
有呢！ 就是用脚手架初始化了个新项目，一句代码也没写。只有 windows 下有问题，mac下我试了是没问题的。
<img width="256" alt="image" src="https://github.com/midwayjs/midway/assets/3368691/8717690e-8b0a-458d-85b5-9a02f59f8f83">

**Reply by czy88840616** [reply]
可能是 ncc 在 windows 底下有 bug ？

**Reply by hooliy** [reply]
也有可能吧，算了还不管它了
